Source/Core/Core/State.cpp

```diff
@@ -554,7 +629,99 @@ u64 GetUnixTimeOfSlot(int slot)
     return 0;
 
   constexpr u64 MS_PER_SEC = 1000;
-  return static_cast<u64>(header.time * MS_PER_SEC) + (DOUBLE_TIME_OFFSET * MS_PER_SEC);
+  return static_cast<u64>(header.legacy_header.time * MS_PER_SEC) +
+         (DOUBLE_TIME_OFFSET * MS_PER_SEC);
+}
+
+static void DecompressLZ4(std::vector<u8>& raw_buffer, u64 size, File::IOFile& f)
+{
+  raw_buffer.resize(size);
+
+  u64 total_bytes_read = 0;
+  while (true)
+  {
+    s32 compressed_data_len;
+    f.ReadArray(&compressed_data_len, 1);
+
+    auto compressed_data = std::make_unique<char[]>(compressed_data_len);
+    f.ReadBytes(compressed_data.get(), compressed_data_len);
+
+    // We need to specify the output buffer's size for safety. This may exceed the positive bound
+    // of int, causing the buffer size to be interpreted as a negative value.
```

Comment:
I don't know what this comment has to do with anything.

Reply:
Yeah this is a bit unnecessary. I think it's clear what the operation below it is doing.